robolectric-shadows/shadows-core/src/main/resources/org/robolectric/fakes/RenderService.java.vm

```diff
@@ -0,0 +1,269 @@
+      // Abort the rendering if the resources are not found.
+      return null;
+    }
+    // find the layout to run
+    ResourceValue value = resourceResolver.getProjectResource(ResourceType.LAYOUT, layoutName);
+    if (value == null || value.getValue() == null) {
+      throw new IllegalArgumentException("layout does not exist");
+    }
+    File layoutFile = new File(value.getValue());
+    ILayoutPullParser parser = null;
+    parser = new XmlParser();
+    parser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, true);
+    parser.setInput(new FileInputStream(layoutFile), "UTF-8"); // $NON-NLS-1$
+    versionAndConstantsSetUp();
+    HardwareConfig hardwareConfigconfig = new HardwareConfig(config.getScreenWidthQualifier().getValue(),
+        config.getScreenHeightQualifier().getValue(), config.getDensityQualifier().getValue(),
+        Xdpi, Ydpi, config.getScreenSizeQualifier().getValue(),
+        config.getScreenOrientationQualifier().getValue(), false);
+    SessionParams p = new SessionParams(parser, renderingMode, this/* projectKey*/, hardwareConfigconfig,
+        resourceResolver, projectCallback, minSdkVersion, targetSdkVersion, logger);
+    params = p;
+    // Request margin and baseline information.
+    // TODO: Be smarter about setting this; start without it, and on the first request
+    // for an extended view info, re-render in the same session, and then set a flag
+    // which will cause this to create extended view info each time from then on in the
+    // same session
+    params.setExtendedViewInfoMode(true);
+    if (!showDecorations) {
+      params.setForceNoDecor();
+    } else {
+      if (appLabel == null) {
+        appLabel = "Random App";
+      }
+      params.setAppLabel(appLabel);
+      params.setAppIcon(appIconName); // ok to be null
+    }
+    if (overrideBgColor != null) {
+      params.setOverrideBgColor(overrideBgColor.intValue());
+    }
+    // set the Image Overlay as the image factory.
+    params.setImageFactory(imageFactory);
+    try {
+      return layoutLib.createSession(params);
+    } catch (RuntimeException t) {
+      // Exceptions from the bridge
+      logger.error(null, t.getLocalizedMessage(), t, null);
+      throw t;
+    }
+  }
+
+  private void versionAndConstantsSetUp() {
+    int size1 = config.getScreenDimensionQualifier().getValue1();
+    int size2 = config.getScreenDimensionQualifier().getValue2();
+    ScreenOrientation orientation = config.getScreenOrientationQualifier().getValue();
+    switch (orientation) {
+      case LANDSCAPE:
+        width = size1 < size2 ? size2 : size1;
+        height = size1 < size2 ? size1 : size2;
+        break;
+      case PORTRAIT:
+        width = size1 < size2 ? size1 : size2;
+        height = size1 < size2 ? size2 : size1;
+        break;
+      case SQUARE:
+        width = height = size1;
+        break;
+    }
+    if (minSdkVersion == -1) {
+      minSdkVersion = config.getVersionQualifier().getVersion();
```

Comment:
I guess not, but if it's not set, defaulting it to the config version makes sense. We don't know if -1 is gonna cause weird behavior.